resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -75,11 +86,13 @@ private[spark] object KubernetesUtils extends Logging {
     }
   }
 
+  @Since("2.3.0")
   def requireNandDefined(opt1: Option[_], opt2: Option[_], errMessage: String): Unit = {
     opt1.foreach { _ => require(opt2.isEmpty, errMessage) }
     opt2.foreach { _ => require(opt1.isEmpty, errMessage) }
   }
 
+  @Since("3.2.0")
   def loadPodFromTemplate(
```

Comment:
This was introduced by https://issues.apache.org/jira/browse/SPARK-24434.
Which went into Spark 3.0.0.

https://github.com/apache/spark/blob/branch-3.0/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala#L82

Reply:
As I mentioned in the PR title, the signature is change recently, @attilapiros . 
I know that because I changed it. :)

Reply:
Please see the following PR description.
> In this PR, Since annotations denote the last function signature changes because these are going to become public at Apache Spark 3.2.0.